packages/react-dom/src/events/DOMEventPriority.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {EventPriority} from 'shared/ReactTypes';
+import type {TopLevelType} from 'legacy-events/TopLevelEventTypes';
+
+import * as DOMTopLevelEventTypes from './DOMTopLevelEventTypes';
+import {DiscreteEvent, UserBlockingEvent, ContinuousEvent} from 'shared/ReactTypes';
+
+// We use a Set here rather than an Array or Object map.
```

Comment:
Out of curiosity: Have you also tried a switch statement?

Reply:
I did and it was even faster, but not by much. The issue with switches was that it was also quite a bit more code size comared to before.